### PR TITLE
Feature/add top series correlation

### PR DESCRIPTION
**Pull request by montsemunoz**

### CODE REVIEW

File: mango/tests/data_module/test_correlation_ts.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    invalid_df6 = None
+    expected_result_invalid_df6 = None
+    invalid_df7 = None
+    expected_result_invalid_df7 = None
+
+    def setUp(self):
+        pass
+
+    @classmethod
+    def setUpClass(cls) -> None:
```
Comment (ggsdc) [inline mango/tests/data_module/test_correlation_ts.py:29]: Why use `setUpClass` instead of `setUp`?

File: requirements.txt [hunk 2]
```diff
@@ -23,4 +23,5 @@ unidecode<=1.3.7
 urllib3<=2.0.7
 XlsxWriter<=3.1.9
 Pyomo<=6.6.2
-Pillow<=10.0.1
\ No newline at end of file
+Pillow<=10.0.1
+tabulate
```
Comment (ggsdc) [inline requirements.txt:27]: Should we also change the `pyproject.toml`?

File: mango/data/__init__.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import pandas
 
 from .calendar_data import get_calendar
+from .series_correlations import get_corr_matrix
```
Comment (gvalleg) [inline mango/data/__init__.py:6]: I think this kind of functions would be better in the processing module, in the processing_time _series.py script.
@AntonioGonzalezSuarez @ggsdc What do you think?

**Review by gvalleg — CHANGES_REQUESTED** [review]
Change the functions to processing module

File: mango/processing/processing_time_series.py [hunk 4]
```diff
@@ -210,3 +212,218 @@ def create_recurrent_dataset(
...
+                raise ValueError("Not all columns in Dataframe are numerics")
+
+
+def get_corr_matrix_aux(
+    df: pd.DataFrame,
+    years_corr,
+    n_top,
+    threshold,
+    subset: List = None,
+) -> Dict[str]:
```
Comment (AntonioGonzalezSuarez) [inline mango/processing/processing_time_series.py:351]: Dict needs two argument. First is the type of the key and second the type of the value. For this case should be as follows:

```python
Dict[str,Dict[str,float]]
```

**Review by AntonioGonzalezSuarez — CHANGES_REQUESTED** [review]

**Review by ggsdc — APPROVED** [review]

**Pull request merged**